stdlib/3/email/feedparser.pyi

```diff
@@ -1,15 +1,21 @@
-# Stubs for email.feedparser (Python 3.4)
-
 from email.message import Message
 from email.policy import Policy
-from typing import Callable
+from typing import Callable, Generic, TypeVar, overload
+
+_M = TypeVar("_M", bound=Message)
 
-class FeedParser:
-    def __init__(self, _factory: Callable[[], Message] = ..., *, policy: Policy = ...) -> None: ...
+class FeedParser(Generic[_M]):
+    @overload
+    def __init__(self: FeedParser[Message], _factory: None = ..., *, policy: Policy = ...) -> None: ...
```

Comment:
Looks like if the policy argument is passed, it gets the factory from `policy.message_factory`, so shouldn't there be a more permissive overload in that case?

https://github.com/python/cpython/blob/aa79707262f893428665ef45b5e879129abca4aa/Lib/email/feedparser.py#L153 for reference.

Reply:
What did you have in mind? As far as I can see, `policy.message_factory` defaults to `None`, so `Message` is used in that case. We could make `Policy` generic as well, possibly.

Reply:
I'm not sure exactly what the right solution is, as I'm not familiar with how this library is actually used. Making Policy also generic risks making users' lives more complicated if in practice people rarely use custom Policies. Perhaps the right call is to leave the simpler version you wrote until we get user complaints from people who use custom message factories, but I wanted to flag this as a possible issue.

Reply:
I haven't used `policy` myself, but I found this in the documentation:

> If policy is specified use the rules it specifies to update the representation of the message. If policy is not set, use the compat32 policy, which maintains backward compatibility with the Python 3.2 version of the email package and provides Message as the default factory. All other policies provide EmailMessage as the default _factory. For more information on what else policy controls, see the policy documentation.

`EmailMessage` is a sub-class of `Message` (via `MIMEPart`). I think this implies that `Message` is the expected bottom here, but it might be worthwhile to add overloads for other built-in policies, or make it generic indeed. Since I have no experience with other policies, I'd not want to do this at this point, though.